src/redux/appSlice.ts

```diff
@@ -0,0 +1,50 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+
+export const LIGHT = 'light';
+export const DARK = 'dark';
+
+interface appState {
+  mode: 'light' | 'dark' | undefined;
+}
+
+export const saveState = (mode: 'light' | 'dark' | undefined) => {
+  try {
+    localStorage.setItem('mode', mode || LIGHT);
+  } catch (error) {
+    console.error('Error saving state:', error);
+  }
+};
+
+const loadState = (): appState => {
+  try {
+    const mode = localStorage.getItem('mode');
+    if (!mode) {
+      return {
+        mode: undefined
+      };
+    }
+    return {
+      mode: mode === DARK ? DARK : LIGHT,
+    };
+  } catch (error) {
+    console.error('Error loading state:', error);
+    return {
+      mode: LIGHT,
+    };
+  }
+};
+
+const initialState: appState = loadState();
+
+const appSlice = createSlice({
```

Comment:
Hmmm... I actually don't think you need a Redux store for theming. You are basically creating another layer of storage, which is not really useful here. `localStorage` should be enough.

You probably need to create a generic React hook `useAppTheme` which will encapsulate and automate everything regarding theming.

Reply:
At first I thought about such a hook, but later I decided to add a new slice to store application configuration that could be expanded in the future. For example to store some user data or some alternative ways to display a list. Maybe it makes sense to leave appSlice?